x-pack/plugins/maps/public/shared/layers/tile_layer.js

```diff
@@ -79,41 +45,37 @@ export class TileLayer extends AbstractLayer {
     }
   }
 
-  async syncLayerWithMB(mbMap) {
+  syncLayerWithMB(mbMap) {
 
     const source = mbMap.getSource(this.getId());
     const mbLayerId = this.getId() + '_raster';
 
-    if (source) {
-      // If source exists, just sync style
-      this._setTileLayerProperties(mbMap, mbLayerId);
-      return;
-    }
+    if (!source) {
+      const sourceDataRquest = this.getSourceDataRequest();
+      const url = sourceDataRquest.getData();
+      if (!url) {
+        return;
+      }
 
-    const sourceDataRquest = this.getSourceDataRequest();
-    const url = sourceDataRquest.getData();
-    if (!url) {
+      const sourceId = this.getId();
+      mbMap.addSource(sourceId, {
+        type: 'raster',
+        tiles: [url],
+        tileSize: 256,
+        scheme: 'xyz',
+      });
+
+      mbMap.addLayer({
+        id: mbLayerId,
+        type: 'raster',
+        source: sourceId,
+        minzoom: this._descriptor.minZoom,
+        maxzoom: this._descriptor.maxZoom,
+      });
       return;
```

Comment:
Nice. Good call returning before a needless call to set layer props

Reply:
oops, that's actually a bug...

Reply:
lol. I was actually just staring at it coming to the same conclusion. we would be doing an extra call to set zoom but alpha isn't covered. there might be a better way to dice those up but a redundant zoom set isn't the end of the world